server/src/main/java/org/apache/druid/server/coordinator/DruidCoordinator.java

```diff
@@ -873,10 +901,14 @@ public void run()
       try {
         final long globalStart = System.nanoTime();
         synchronized (lock) {
-          if (!coordLeaderSelector.isLeader()) {
-            log.info("LEGGO MY EGGO. [%s] is leader.", coordLeaderSelector.getCurrentLeader());
+          DruidLeaderSelector.LeaderState leaderState = coordLeaderSelector.isLeader();
+          if (leaderState == DruidLeaderSelector.LeaderState.NOT_ELECTED) {
+            log.info("I am no longer the leader. Current leader is [%s].", coordLeaderSelector.getCurrentLeader());
             stopBeingLeader();
             return;
+          } else if (leaderState == DruidLeaderSelector.LeaderState.ELECTED) {
+            throw new ISE(
```

Comment:
Better to just log an error here and return.
An exception thrown here is just caught, logged and swallowed in `ScheduledExecutors.scheduledAtFixedRate()`.